core/src/backend/video/software.rs

```diff
@@ -433,3 +434,202 @@ mod vp6 {
         }
     }
 }
+
+//#[cfg(feature = "screenvideo")]
+mod screen {
+    // This module is heavily based on flashsv.rs from NihAV,
+    // written by Kostya Shishkov, with permission.
+
+    use crate::backend::video::software::VideoDecoder;
+    use crate::backend::video::{DecodedFrame, EncodedFrame, Error, FrameDependency};
+
+    use flate2::Decompress;
+
+    /// Screen Video (V1 only) decoder.
+    pub struct ScreenVideoDecoder {
```

Comment:
I think this deserves to be in a separate file? :)

Reply:
I was _just_ about to comment that since the _entire_ decoder is smaller than just the glue for the VP6 one, it might not be worth the hassle... :D

Reply:
Of course, I can split into a module under a `decoders` folder (similar to how it's done for audio).
But then I'd do the same with the other two as well.
Also, these are only the decoders for the `software` backend... - not that we have anything else at the moment.

Reply:
So, how about something like this? [see latest commit]